powerauth-docs/test-vectors/signatures-offline.json

```diff
@@ -0,0 +1,3124 @@
+{
+  "description" : "Client must be able to compute PowerAuth offline signature (using 1FA, 2FA signature keys) based on given data, counter and signature type",
+  "data" : [ {
+    "input" : {
+      "signaturePossessionKey" : "PTCBpEsGknUzlYG0Y9dumQ==",
+      "signatureKnowledgeKey" : "sGfvJAi5h6sxXV+vEG/2iQ==",
+      "signatureBiometryKey" : "otzFKCko5lcEGRev//uEbQ==",
+      "signatureType" : "possession",
+      "signatureComponentLength" : "2",
```

Comment:
I find this implementation misleading.

Why do we allow `signatureComponentLength` of `2`, when actual the signature length is `4`.

I think we should reject such configuration to avoid surprises in cryptography code and allow only lengths 4 - 12, as expected.

Reply:
I found the same issue. Also there's a problem that the test generator doesn't generate vectors for the default signatureComponentLength equal to 8

Reply:
@hvge @romanstrobl The conversion methods currently do not throw any exceptions or return error codes. As a result, we might break the behavior (what happens if we randomly throw an exception somewhere before computing the signature?). Submitting the length 2, 3, -1 and obtaining length 4 is legitimate behavior we can proxy on other levels (i.e., in PowerAuth Server or PowerAuth Cloud).

Reply:
Fixed.